tools/swift-api-digester/ModuleAnalyzerNodes.cpp

```diff
@@ -1358,6 +1358,29 @@ SwiftDeclCollector::constructTypeDeclNode(NominalTypeDecl *NTD) {
   return TypeNode;
 }
 
+/// Checks if any of the conformances in the provided type decl come from any
+/// of the extensions passed in. This is necessary because we don't want to
+/// show any types in the SDK dump if they are merely extended with internal
+/// members, but we definitely do want to show the types if they declare
+/// any public conformances.
+static bool anyConformancesComeFromExtensions(const SDKContext &Ctx,
+  ArrayRef<ExtensionDecl *> Exts, const NominalTypeDecl *NTD) {
+  llvm::SmallPtrSet<ExtensionDecl *, 3> conformanceExtensions;
+
+  for (auto *conf : NTD->getAllConformances()) {
```

Comment:
This is a pretty expensive operation. Parseable interface printing goes the other direction, by looking at an extension and seeing if it introduces any interesting conformances, directly or indirectly.

…hang on, don't we have to emit the conformances themselves somewhere? Why wouldn't this logic go there?

Reply:
The SDK attaches the declared conformances as a child of the type decl itself.

```json
      "conformances": [
        {
          "kind": "Conformance",
          "name": "MyProto",
          "printedName": "MyProto"
        }
      ]
```

It looks like the conformance table is cached -- this call here is always called after a call to `getAllConformances()` has already happened.

I'm happy to flip this the other direction, though.

Reply:
In that case, can't you just check that that table is empty/absent?

Reply:
Well, the table could also contain conformances declared in the original module. I only care if _this module_ added conformances to public types.

Reply:
Sorry, by "this table" I meant the JSON node that's going to get emitted under "conformances".

Reply:
Yep, that table also contains conformances declared in the original module.

Reply:
Oh. That seems…wrong. What if *another* module adds a new conformance?